src/main/java/ru/hogwarts/school/repository/FacultyRepository.java

```diff
@@ -4,7 +4,13 @@
 import ru.hogwarts.school.model.Faculty;
 
 import java.util.Collection;
+import java.util.Optional;
 
 public interface FacultyRepository extends JpaRepository<Faculty, Long> {
-    Collection<Faculty> findByColor(String color);
+    Collection<Faculty> findByColorIgnoreCase(String color);
+
+    Optional<Faculty> findFirstByNameOrColorIgnoreCase(String name, String color);
```

Comment:
```suggestion
    List<Faculty> findAllByNameIgnoreCaseOrColorIgnoreCase(String name, String color);
```
Во-первых, вернуть надо список, во-вторых, игнорировать регистр надо для имени тоже, а не только для цвета